components/director/internal/features/features.go

```diff
@@ -4,7 +4,7 @@ package features
 type Config struct {
 	DefaultScenarioEnabled        bool   `envconfig:"default=true,APP_DEFAULT_SCENARIO_ENABLED"`
 	ProtectedLabelPattern         string `envconfig:"default=.*_defaultEventing|^consumer_subaccount_ids$"`
-	ImmutableLabelPattern         string `envconfig:"default=xsappnameCMPClone,APP_SELF_REGISTER_LABEL_KEY"`
+	ImmutableLabelPattern         string `envconfig:"default=^xsappnameCMPClone$,APP_SELF_REGISTER_LABEL_KEY"`
```

Comment:
The way it's written now, we'll get the "immutable" pattern from `APP_SELF_REGISTER_LABEL_KEY` and if there isn't such environment variable we'll use default value which is `^xsappnameCMPClone$` right? But `APP_SELF_REGISTER_LABEL_KEY` is configure to `xsappnameCMPClone` not `^xsappnameCMPClone$`. Maybe it's better to create a new environment variable dedicated to immutable pattern which value will be the value from `APP_SELF_REGISTER_LABEL_KEY` but with `^` in the begging and `$` in the end and remove the default value? That way if for some reason we change the label key it should work out of the box. Because the immutable pattern is that label key but with the special symbols